presto-proxy/src/main/java/com/facebook/presto/proxy/ProxyResource.java

```diff
@@ -296,10 +296,10 @@ private static boolean isPrestoHeader(String name)
 
     private static Response responseWithHeaders(ResponseBuilder builder, ProxyResponse response)
     {
-        response.getHeaders().asMap().forEach((headerName, value) -> {
+        response.getHeaders().asMap().forEach((headerName, values) -> {
```

Comment:
Would  `response.getHeaders().forEach((headerName, value)` work? Just remove `asMap()`.

Reply:
That works.